### PR TITLE
fix(ad_api_adaptors): fix to merge waypoint

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Fix the bug where consecutive waypoints would be ignored.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp [hunk 1]
```diff
@@ -29,29 +29,57 @@ RoutingAdaptor::RoutingAdaptor() : Node("routing_adaptor")
...
-  route_points_ = std::make_shared<SetRoutePoints::Service::Request>();
+  adaptor.init_sub(
+    sub_state_, [this](const RouteState::Message::ConstSharedPtr msg) { state_ = msg->state; });
+
+  const auto rate = rclcpp::Rate(5.0);
+  timer_ = rclcpp::create_timer(
+    this, get_clock(), rate.period(), std::bind(&RoutingAdaptor::on_timer, this));
+
+  state_ = RouteState::Message::UNKNOWN;
+  route_ = std::make_shared<SetRoutePoints::Service::Request>();
```
Comment (takayuki5168) [inline system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp:40]: Setting nullptr first sounds better to me.
If you don't think it's good, just ignore please.

File: system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp [hunk 2]
```diff
@@ -29,29 +29,57 @@ RoutingAdaptor::RoutingAdaptor() : Node("routing_adaptor")
...
-  route_points_->goal = pose->pose;
-  route_points_->waypoints.clear();
+  // Wait a moment to merge waypoints. This value is used for flag and delay counter.
+  constexpr int delay_count = 3;
+  if (0 < request_control_ && request_control_ < delay_count) {
+    ++request_control_;
+  }
+  if (request_control_ != delay_count) {
+    return;
+  }
```
Comment (takayuki5168) [inline system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp:52]: I don't understand this algorithm well.
Why 3 is chosen?

What does `merge waypoints` mean.

`request_control_` seems to have additional meaning, not only what the variable name expresses.
Could you write a comment of what `request_control_` means.

File: system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp (hunk 1 above)
Comment (isamu-takagi) [reply system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp:40]: This value is always used and does not need to be nullptr.

File: system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp (hunk 2 above)
Comment (isamu-takagi) [reply system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp]: Done https://github.com/autowarefoundation/autoware.universe/pull/2215/commits/343a3572982dbbabbde4035b279d937b8b816e10.

File: system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp (hunk 1 above)
Comment (takayuki5168) [reply system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp:40]: Then why do you make route_ shared pointer?

**Review by takayuki5168 — APPROVED** [review]
LGTM

File: system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp (hunk 1 above)
Comment (isamu-takagi) [reply system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp:40]: Because the `async_send_request` function only supports shared pointer.

Comment (takayuki5168) [reply system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp:40]: Got it. Thank you

**Pull request merged**